hpke.go

```diff
@@ -256,12 +320,29 @@ func (ctx *EncryptContext) Seal(aad, pt []byte) []byte {
 	return ct
 }
 
+func (ctx *EncryptContext) Marshal() ([]byte, error) {
+	return syntax.Marshal(ctx.inner)
+}
+
+func UnmarshalEncryptContext(opaque []byte) (*EncryptContext, error) {
+	ctx, err := unmarshalCipherContext(opaque)
+	if err != nil {
+		return nil, err
+	}
+
+	if ctx.inner.Role != roleSender {
```

Comment:
I would push this up into `unmarshalCipherContext`, so you would have `unmarshalCipherContext(data []byte, role Role)` and call it as `unmarshalCipherContext(data, roleSender)`.

Reply:
Done, except the role is the first parameter. This aligns with `newCipherContext`.